spark-connector/src/main/java/com/datastrato/gravitino/spark/table/SparkBaseTable.java

```diff
@@ -0,0 +1,129 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.spark.table;
+
+import com.datastrato.gravitino.spark.ConnectorConstants;
+import com.datastrato.gravitino.spark.PropertiesConverter;
+import com.datastrato.gravitino.spark.SparkTypeConverter;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.spark.sql.catalyst.analysis.NoSuchTableException;
+import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.connector.catalog.SupportsRead;
+import org.apache.spark.sql.connector.catalog.SupportsWrite;
+import org.apache.spark.sql.connector.catalog.Table;
+import org.apache.spark.sql.connector.catalog.TableCapability;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.connector.read.ScanBuilder;
+import org.apache.spark.sql.connector.write.LogicalWriteInfo;
+import org.apache.spark.sql.connector.write.WriteBuilder;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.MetadataBuilder;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.sql.types.StructType$;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
+
+/**
+ * Provides schema info from Gravitino, IO from the internal spark table. The specific catalog table
+ * could implement more capabilities like SupportsPartitionManagement for Hive table, SupportsIndex
+ * for JDBC table, SupportsRowLevelOperations for Iceberg table.
+ */
+public class SparkBaseTable implements Table, SupportsRead, SupportsWrite {
+  private Identifier identifier;
+  private com.datastrato.gravitino.rel.Table gravitinoTable;
+  private TableCatalog sparkCatalog;
+  private Table lazySparkTable;
+  private PropertiesConverter propertiesConverter;
+
+  public SparkBaseTable(
+      Identifier identifier,
+      com.datastrato.gravitino.rel.Table gravitinoTable,
+      TableCatalog sparkCatalog,
+      PropertiesConverter propertiesConverter) {
+    this.identifier = identifier;
+    this.gravitinoTable = gravitinoTable;
+    this.sparkCatalog = sparkCatalog;
+    this.propertiesConverter = propertiesConverter;
+  }
+
+  protected Table getSparkTable() {
+    if (this.lazySparkTable == null) {
+      try {
+        this.lazySparkTable = sparkCatalog.loadTable(identifier);
+      } catch (NoSuchTableException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    return this.lazySparkTable;
```

Comment:
Adding `this.` only when the class variable is assigned/initialized.

Reply:
done